pkg/kubelet/dockershim/docker_container.go

```diff
@@ -217,13 +221,34 @@ func (ds *dockerService) createContainerLogSymlink(containerID string) error {
 	if err != nil {
 		return fmt.Errorf("failed to get container %q log path: %v", containerID, err)
 	}
+
 	if path != "" {
-		// Only create the symlink when container log path is specified.
-		if err = ds.os.Symlink(realPath, path); err != nil {
-			return fmt.Errorf("failed to create symbolic link %q to the container log file %q for container %q: %v",
-				path, realPath, containerID, err)
+		if realPath != "" {
+			// Only create the symlink when container log path is specified and log file exists.
+			if err = ds.os.Symlink(realPath, path); err != nil {
+				return fmt.Errorf("failed to create symbolic link %q to the container log file %q for container %q: %v",
+					path, realPath, containerID, err)
+			}
+		} else {
+			dockerLoggingDriver := ""
+			dockerInfo, err := ds.client.Info()
+			if err != nil {
+				glog.Errorf("Failed to execute Info() call to the Docker client: %v", err)
+			} else {
+				dockerLoggingDriver = dockerInfo.LoggingDriver
+				glog.V(10).Infof("Docker logging driver is %s", dockerLoggingDriver)
+			}
+
+			if dockerLoggingDriver == dockerDefaultLoggingDriver {
+				glog.Errorf("Cannot create symbolic link because container log file doesn't exist!")
+			} else {
+				glog.V(5).Infof("Unsupported logging driver: %s", dockerLoggingDriver)
+			}
 		}
+	} else {
+		glog.V(5).Infof("Container %s log path isn't specified, will not create the symlink", containerID)
```

Comment:
Sorry for the nit picky... However, it would be better to move this above to reduce unnecessary indent.

```
If path == "" {
// log
return nil
}
// The following block
}
```

Otherwise LGTM